File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/adapter/transport/AirTransportAdapter.java
package com.bigboxcode.designpattern.adapter.transport;

public class AirTransportAdapter implements Transport {
    private final AirTransport airTransport;

    public AirTransportAdapter(AirTransport airTransport) {
        this.airTransport = airTransport;
    }

    @Override
    public int getNumberOfWheels() {
        return this.airTransport.getNumberOfWheels();
    }

    @Override
    public double getWeight() {
        return this.airTransport.getWeight();
    }

    @Override
    public double getDistanceTravelled() {
        // Convert nautical mile to mile and return
        double distanceInNauticalMile = this.airTransport.getDistanceTravelled();

        return distanceInNauticalMile * 1.151;
    }

    @Override
    public double getTravelCostPerMile() {
        // calculate cost per mile from total cost
        double totalCost = this.airTransport.getTravelCostTotal();

        return totalCost / this.getDistanceTravelled();
    }
}

File: design-patterns/php/adapter/transportadapter/AirTransportAdapter.php
<?php
// AirTransportAdapter.php

namespace BigBoxCode\DesignPattern\Adapter\TransportAdapter;


class AirTransportAdapter implements Transport {
    public function __construct(private AirTransport $airTransport) {
        $this->airTransport = $airTransport;
    }

    public function getNumberOfWheels(): int {
        return $this->airTransport->getNumberOfWheels();
    }

    public function getWeight(): float {
        return $this->airTransport->getWeight();
    }

    public function getDistanceTravelled(): float {
        $distanceInNauticalMile = $this->airTransport->getDistanceTravelled();
        return $distanceInNauticalMile * 1.151;
    }

    public function getTravelCostPerMile(): float {
        $totalCost = $this->airTransport->getTravelCostTotal();
        return $totalCost / $this->getDistanceTravelled();
    }
}
